pkg/controller/sshd/sshd_controller.go

```diff
@@ -0,0 +1,564 @@
+		copiedService.Spec.HealthCheckNodePort = foundService.Spec.HealthCheckNodePort
+		if !reflect.DeepEqual(foundService.Spec, copiedService.Spec) {
+			// Specs aren't equal, update and fix.
+			r.SetSSHDStatusPending(instance, reqLogger, "Updating service", "from", foundService.Spec, "to", service.Spec)
+			foundService.Spec = *copiedService.Spec.DeepCopy()
+			if err = r.client.Update(context.TODO(), foundService); err != nil {
+				r.SetSSHDStatusError(instance, reqLogger, "Failed to update service", err)
+				return reconcile.Result{}, err
+			}
+			// Requeue to give AWS time to apply the changes.
+			reqLogger.Info("Requeuing after service update")
+			return reconcile.Result{Requeue: true, RequeueAfter: 10 * time.Second}, nil
+		}
+	}
+
+	// Create a Route53 DNS entry for the service's load balancer.
+	// FIXME Consider using https://github.com/openshift/external-dns
+
+	clusterBaseDomain, err := utils.GetClusterBaseDomain(r.client)
+	if err != nil {
+		r.SetSSHDStatusError(instance, reqLogger, "Failed to get cluster's base domain", err)
+		return reconcile.Result{}, err
+	}
+
+	// Get the ELB-to-be's name from Service's UID.
+	elbName := strings.ReplaceAll("a"+string(foundService.ObjectMeta.UID), "-", "")
+	if len(elbName) > 32 {
+		// truncate to 32 characters
+		elbName = elbName[0:32]
+	}
+
+	exists, elb, err := r.awsClient.DoesELBExist(elbName)
+	if err != nil {
+		r.SetSSHDStatusError(instance, reqLogger, "Failed to get load balancer status", err)
+		return reconcile.Result{}, err
+	}
+	if !exists {
+		// It isn't bad that it doesn't exist if there's no error, so re-queue.
+		r.SetSSHDStatusPending(instance, reqLogger, "Waiting on service load balancers")
+		return reconcile.Result{Requeue: true, RequeueAfter: 10 * time.Second}, nil
+	}
+
+	r.route53 = &Route53Data{
+		loadBalancerDNSName:      elb.DNSName,
+		loadBalancerHostedZoneId: elb.DNSZoneId,
+		resourceRecordSetName:    instance.Spec.DNSName + "." + clusterBaseDomain,
+		privateHostedZoneName:    clusterBaseDomain + ".",
+		// The public zone name omits the cluster name.
+		// e.g. mycluster.abcd.s1.openshift.com -> abcd.s1.openshift.com
+		publicHostedZoneName: clusterBaseDomain[strings.Index(clusterBaseDomain, ".")+1:] + ".",
+	}
+
+	err = r.ensureDNSRecords()
+	if err != nil {
+		r.SetSSHDStatusError(instance, reqLogger, "Failed to ensure the DNS record", err)
+		return reconcile.Result{}, err
+	}
+
+	r.SetSSHDStatus(instance, "SSHD is ready", cloudingressv1alpha1.SSHDStateReady)
+
+	return reconcile.Result{RequeueAfter: 60 * time.Second}, nil
+}
+
+func getMatchLabels(cr *cloudingressv1alpha1.SSHD) map[string]string {
+	return map[string]string{"deployment": cr.Name}
+}
+
+func newSSHDDeployment(cr *cloudingressv1alpha1.SSHD, configMapList *corev1.ConfigMapList) *appsv1.Deployment {
+	replicas := int32(1)
+	terminationGracePeriod := int64(30) // seconds
+	revisionHistoryLimit := int32(10)
+	progressDeadline := int32(600) // seconds
```

Comment:
Please consider using `"k8s.io/utils/pointer"` and then in-place `pointer.Int32ptr(1)`, pointer.Int64ptr(30)` and so on.

Reply:
Oh nice!  Didn't know about that package.